src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -713,11 +713,19 @@ export class SearchResult extends Disposable {
 		this.disposeMatches();
 	}
 
-	remove(matches: FileMatch | FileMatch[]): void {
+	remove(matches: FileMatch | FolderMatch | (FileMatch | FolderMatch)[]): void {
 		if (!Array.isArray(matches)) {
 			matches = [matches];
 		}
 
+		matches.forEach(m => {
+			if (m instanceof FolderMatch) {
+				m.clear();
+			}
+		});
+
+		matches = matches.filter(m => m instanceof FileMatch);
+
```

Comment:
It seems `[FileMatch, FolderMatch]` is not handled?

Reply:
https://github.com/microsoft/vscode/pull/78518/files#diff-fb27c71af255c55c348da6ee665d2d80R717 handled it.